src/repositories/productRepo.js

```diff
@@ -9,12 +10,23 @@ const save = (product) => {
 
 const findById = (id) => Product.findById(id);
 
+
+const deleteProductAndReviews = (productId) => {
+  
+       Product.findByIdAndDelete(productId);
+  
+       Review.deleteMany({ productId: productId });
```

Comment:
1. Do we have a `productId` in our reviews?
2. Is this working on your local?
3. And why are we not waiting for the promise to get resolved.
4. Check and try running on your local there seems to a wrong logic being applied here.